src/trans.rs

```diff
@@ -914,9 +964,16 @@ impl<'v, 'tcx: 'v> BinaryenFnCtxt<'v, 'tcx> {
 
     fn trans_lval(&mut self, lvalue: &Lvalue<'tcx>) -> BinaryenLvalue {
         let i = match *lvalue {
-            Lvalue::Arg(i) => i,
-            Lvalue::Var(i) => self.mir.arg_decls.len() as u32 + i,
+            Lvalue::Arg(i) => {
+                let i: u32 = unsafe { mem::transmute(i) };
+                i
+            },
+            Lvalue::Var(i) => {
+                let i: u32 = unsafe { mem::transmute(i) };
+                self.mir.arg_decls.len() as u32 + i
+            }
             Lvalue::Temp(i) => {
+                let i: u32 = unsafe { mem::transmute(i) };
```

Comment:
Fair.  I had a hell of a time finding the definition of the `Temp` struct (in this case).  I think I was just taking out my frustration on the code. :)

I just pushed a fix to use `i.index()` instead.  Is that what you mean by `local_index` - or is there an even better way to simplify this?